mixnet/mixnet.py

```diff
@@ -1,19 +1,44 @@
 from __future__ import annotations
 
-import random
-from dataclasses import dataclass
-from typing import List
+import asyncio
+from typing import Self
 
+from cryptography.hazmat.primitives.asymmetric.x25519 import (
+    X25519PrivateKey,
+)
+
+from mixnet.client import MixClient
 from mixnet.node import MixNode
+from mixnet.topology import MixnetTopology
 
 
 class Mixnet:
-    __topology: MixnetTopology | None = None
-
-    def get_topology(self) -> MixnetTopology:
-        if self.__topology is None:
-            raise RuntimeError("topology is not set yet")
-        return self.__topology
+    __mixclient: MixClient
+    __mixnode: MixNode
+
+    @classmethod
+    async def new(
+        cls,
+        initial_topology: MixnetTopology,
+        emission_rate_per_min: int,  # Poisson rate parameter: lambda
+        redundancy: int,
+        encryption_private_key: X25519PrivateKey,
+        delay_rate_per_min: int,  # Poisson rate parameter: mu
```

Comment:
Also the `delay_rate_per_min` could be provided by the robustness layer.

Reply:
Resolved by https://github.com/logos-co/nomos-specs/pull/57#discussion_r1469009270